File: src/main/java/com/samyyc/setu/listener/SetuListener.java
package com.samyyc.setu.listener;

import com.samyyc.setu.util.MessageUtil;
import net.mamoe.mirai.event.GlobalEventChannel;
import net.mamoe.mirai.event.events.MessageEvent;
import net.mamoe.mirai.message.data.*;

import java.util.List;
import java.util.stream.Collectors;

public class SetuListener {

    public static void listen() {
        GlobalEventChannel.INSTANCE.subscribeAlways(MessageEvent.class, event -> {
            MessageChain chain = event.getMessage();
            List<SingleMessage> messages = chain.stream().filter(singleMessage -> !(singleMessage instanceof MessageSource)).collect(Collectors.toList());
            if (messages.size() > 0) {
                if (messages.get(0) instanceof PlainText) {
                    PlainText plainText = (PlainText) messages.get(0);
                    String message = plainText.contentToString();
                    if (message.startsWith("色图")) {
                        event.getSubject().sendMessage(MessageUtil.getSetuMessageFromApi(false, event.getSubject(), message));
                    } else if (message.startsWith("r18色图")) {
                        event.getSubject().sendMessage(MessageUtil.getSetuMessageFromApi(true, event.getSubject(), message));
                    }
                }
            }
        });
    }

}

File: src/main/kotlin/github/samyycx/miraisetu/listener/SetuListener.kt
package github.samyycx.miraisetu.listener

import github.samyycx.miraisetu.MiraiSetu
import github.samyycx.miraisetu.sender.SetuSender
import net.mamoe.mirai.event.GlobalEventChannel
import net.mamoe.mirai.event.events.GroupMessageEvent
import net.mamoe.mirai.event.events.MessageEvent
import net.mamoe.mirai.event.globalEventChannel
import net.mamoe.mirai.message.data.PlainText

object SetuListener {

    inline fun <reified T : MessageEvent> subscribe() {
        GlobalEventChannel.subscribeAlways<T> { event ->
            if (event.message[1] is PlainText) {
                val plainText = event.message[1] as PlainText
                val text = plainText.contentToString()
                if (text.startsWith("色图")) {
                    val reply = SetuSender.getSetuMessage(text, event.subject)
                    println(3)
                    event.subject.sendMessage(reply)
                } else {
                    println(text)
                    println(event.message.size)
                }
            }
        }
    }

}